### ISSUE TITLE
System.ArgumentOutOfRangeException

### ISSUE THREAD

**Issue by fvet** [StefanMaron/BusinessCentral.LinterCop#330]
When running v0.1.6, I get following errors when loading a project.

```
Analyzer 'BusinessCentral.LinterCop.Design.Rule0021BuiltInMethodImplementThroughCodeunit' threw an exception of type 'System.ArgumentOutOfRangeException' with message 'System.ArgumentOutOfRangeException: Index and length must refer to a location within the string. (Parameter 'length')
   at System.String.Substring(Int32 startIndex, Int32 length)
   at BusinessCentral.LinterCop.Design.Rule0021BuiltInMethodImplementThroughCodeunit.CheckBuiltInMethod(OperationAnalysisContext ctx)
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass53_1.<ExecuteOperationAction>b__1() in D:\a\_work\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 766
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info) in D:\a\_work\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 1102'
```

```
Analyzer 'BusinessCentral.LinterCop.Design.Rule0021BuiltInMethodImplementThroughCodeunit' threw an exception of type 'System.ArgumentOutOfRangeException' with message 'System.ArgumentOutOfRangeException: Index and length must refer to a location within the string. (Parameter 'length')
   at System.String.Substring(Int32 startIndex, Int32 length)
   at BusinessCentral.LinterCop.Design.Rule0021BuiltInMethodImplementThroughCodeunit.CheckBuiltInMethod(OperationAnalysisContext ctx)
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass53_1.<ExecuteOperationAction>b__1() in D:\a\_work\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 766
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info) in D:\a\_work\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 1102'
```

**Comment by Arthurvdv** [comment]
There where no changes on this rule, so kinda surprised you have an error on this one.

Do you have a sample AL file which reproduces this error?

**Comment by fvet** [comment]
Unfortunately not (it's a 600+ files app ...)

**Comment by Arthurvdv** [comment]
I think I know where the issue occurs. I'll try to see what I can do.

**Comment by tscottjendev** [comment]
Having the same issue.

**Comment by Arthurvdv** [comment]
@fvet and @tscottjendev, I've created an fix for this issue and released this in the prerelease.

Could you (briefly) switch over to the prelease version of the LinterCop and provide feedback if this solves your issue?

![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/44637996/7f9a5a85-9de3-4462-84cd-7b623ba0824a)
(not sure of you need to run the `LC: Download Linter Cop` command again to retrieve te latest version)

...or manually download the [BusinessCentral.LinterCop.current.dll](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.5) and rename this to `BusinessCentral.LinterCop.dll` and copy over the file in `C:\Users\<username>\.vscode\extensions\ms-dynamics-smb.al-<version>\bin\Analyzers\`

**Comment by fvet** [comment]
@Arthurvdv Running on prerelease version now (using the 'load pre releases' setting + Download Lintercop), but still have similar warnings.

Might be related to the Page.run(0, ...)?

![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/25268332/473eb102-21d1-4ae3-a26c-9294e4f22079)

(In VSCode, I don't see the warnings, while In DevOps it get the warnings)

![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/25268332/d3d43542-c666-4c9c-9f17-35a951f311c2)

**Comment by fvet** [comment]
Fixing the Page.Run(0, ...) code (LC0027 warning) makes the app.json "Analyzer 'BusinessCentral.LinterCop.Design.Rule0021BuiltInMethodImplementThroughCodeunit' threw an exception of type 'System.ArgumentOutOfRangeException' with message 'System.ArgumentOutOfRangeException: Index and length must refer to a location within the string." warning dissappear

**Comment by tscottjendev** [comment]
The errors have been resolved for me.  I have some the LC0029 `Page.Run(0, <recordVar>)` and no errors.  But I don't have any other `Page.Run(<pageID>, <recordVar>)` warnings (that are not pragma'd out) and since then no errors.

**Comment by fvet** [comment]
@Arthurvdv Just wondering if the LC0027 rule is not too strict about `Page.Run(0, <recordVar>)`. (having pageid set to 0)
Aren't these valid constructs too, as they do not impose a fix page id, resulting in BC internally resolving the page id anyways via the pagemanagement functions?

**Comment by Arthurvdv** [comment]
I've released a new version [v0.30.5](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.5) which should resolve this error.

@tscottjendev and @fvet , I've also included the exception for `Page.Run(0, <recordVar>)` in this release.

Could you provide feedback if the new version resolves this issue?

**Comment by Arthurvdv** [comment]
I’m going to assume this issue is resolved.

If you’re still are experiencing problems on this, feel free to re-open this issue.
